### PR TITLE
The transition between the cubic powerloss and linear powerloss for the supermatter is now smooth.

### PR DESCRIPTION
**Pull request by Pickle-Coding**
## About The Pull Request
Change the requirement for the linear powerloss to occur when the derivative of the cubic powerloss is equal to the rate of the linear powerloss. Offsets linear powerloss to make the transition between the two functions completely smooth. Adds defines for the powerloss magic numbers. Changes the powerloss inhibitor stuff to just reduce the powerloss instead of reducing the functions, meaning that reducing it by 90% would always reduce it by 90%.

## Why It's Good For The Game
This is what powerloss currently looks like (black is power, red is powerloss): image

What it looks like with this PR's changes: image

Now to show why I made this PR, this is what power after powerloss gets applied looks like (black is power, purple is power after powerloss gets applied): image

What it looks like with the new changes in this PR: image

Well, the consequence of powerloss being a cubic function means at one point the derivative of power minus powerloss will be negative, and at one insane point, this could make adding more emitters (or extremely high temperature) actually cause the SM to have lower power than if they didn't add emitters (or extremely high temperature), because linear powerloss doesn't occur in time.

This should make the powerloss function for high power supermatters to be more intuitive, where powerloss doesn't punish giving the supermatter high power to the point of being worse than giving it lower power. Changing powerloss inhibition related stuff will make it more intuitive for people to understand.

## Changelog

:cl:
balance: Changes supermatter powerloss function. It will transition to linear powerloss function at 5.88076GeV, and the linear powerloss function has been offset so that the transition between the two functions is completely smooth.
balance: Changes powerloss inhibition (both CO2 and psychologist effect) to change the rate of powerloss instead of changing the rate of both functions then comparing them to which one should be used.
/:cl:

### CODE REVIEW

**Comment by Pickle-Coding** [comment]
This is what powerloss use to look like (black is power, red is powerloss): image

What it looks like now: image

Now to show why I made this PR, this is what power after powerloss gets applied use to look like (black is power, purple is power after powerloss gets applied): image

What it looks like now: image

**Comment by itseasytosee** [comment]
Best PR all month. How many times differentiable is it?

**Comment by Pickle-Coding** [comment]
Technically 0 because the defines aren't infinitely precise, so there might be a very very small jump.

File: code/modules/power/supermatter/supermatter.dm [hunk 1]
```diff
@@ -266,6 +266,8 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 	var/cascade_initiated = FALSE
 	///Reference to the warp effect
 	var/atom/movable/supermatter_warp_effect/warp
+	///How much power the supermatter is losing.
+	var/power_loss = 0
```
Comment (LemonInTheDark) [inline code/modules/power/supermatter/supermatter.dm:270]: why does this need to be a var on the object?

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]
So high power supermatters will be easier to retain and pump up?
How significant of a difference does that make?

File: code/modules/power/supermatter/supermatter.dm (hunk 1 above)
Comment (Pickle-Coding) [reply code/modules/power/supermatter/supermatter.dm:270]: I kinda forgot why. Maybe for vv, or to allow someone to use it such as for NT CIMS without having to add it.

**Comment by Pickle-Coding** [comment]
Has no impact on anything below 5.88076GeV, but it's quite noticeable in the region above that but not insanely high.

This would make the powerloss for a 10GeV SM be equal to 5GeV instead of 8GeV, bringing it down to around 5GeV instead of 2GeV.

The powerloss is still very high and retaining it is very very hard, and wouldn't last long if you can't retain it. This just prevents the scenario where adding more emitters in an engine with very high power (but no CO2) actually causes the SM to have less power.

**Comment by LemonInTheDark** [comment]
I'll trust you on that, sure

**Pull request merged**